src/test/java/io/jenkins/plugins/analysis/warnings/recorder/RecorderITest.java

```diff
@@ -0,0 +1,230 @@
+package io.jenkins.plugins.analysis.warnings.recorder;
+
+import java.util.List;
+
+import org.junit.Test;
+
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+
+import hudson.model.FreeStyleBuild;
+import hudson.model.FreeStyleProject;
+import hudson.model.Result;
+
+import io.jenkins.plugins.analysis.core.model.AnalysisResult;
+import io.jenkins.plugins.analysis.core.steps.IssuesRecorder;
+import io.jenkins.plugins.analysis.core.testutil.IntegrationTestWithJenkinsPerSuite;
+import io.jenkins.plugins.analysis.core.util.QualityGate.QualityGateResult;
+import io.jenkins.plugins.analysis.core.util.QualityGate.QualityGateType;
+import io.jenkins.plugins.analysis.core.util.QualityGateStatus;
+import io.jenkins.plugins.analysis.warnings.Java;
+import io.jenkins.plugins.analysis.warnings.recorder.pageobj.InfoPage;
+
+import static io.jenkins.plugins.analysis.core.assertions.Assertions.*;
+
+/**
+ * Integration Tests for the IssueRecorder Class.
+ */
+public class RecorderITest  extends IntegrationTestWithJenkinsPerSuite {
+    /**
+     * Test that a freestyle job passes successfully. Quality gate should not negatively affect the result.
+     */
+    @Test
+    public void shouldPassFreestyleJobWithJavaWarnings() {
+        FreeStyleProject project = createFreeStyleProject();
+
+        Java java = new Java();
+        java.setPattern("**/*.txt");
+        IssuesRecorder issuesRecorder = enableWarnings(project, java);
+
+        issuesRecorder.addQualityGate(5, QualityGateType.TOTAL, QualityGateResult.FAILURE);
+
+
+        copySingleFileToWorkspace(project, "../javac.txt", "javac.txt");
+
+        AnalysisResult result = scheduleBuildAndAssertStatus(project, Result.SUCCESS);
+
+
+        assertThat(result).hasTotalSize(2);
+        assertThat(result).hasQualityGateStatus(QualityGateStatus.PASSED);
+    }
+
+    /**
+     * Test that a freestyle job fails if quality gate is passed.
+     */
+    @Test
+    public void shouldFailFreestyleJobWithJavaWarnings() {
+        FreeStyleProject project = createFreeStyleProject();
+
+        Java java = new Java();
+        java.setPattern("**/*.txt");
+        IssuesRecorder issuesRecorder = enableWarnings(project, java);
```

Comment:
Test Setup am besten in Methoden auslagern, dann kann es in anderen Tests wiederverwendet werden.